src/models/challenges.d.ts

```diff
@@ -66,3 +66,31 @@ declare namespace socket_api {
         }
     }
 }
+
+declare namespace rest_api {
+    // Payload of challenge POST
+    interface ChallengeDetails {
+        initialized: boolean;
+        min_ranking: number;
+        max_ranking: number;
+        challenger_color: string; // “automatic”
```

Comment:
Are the comments for default values still meaningful in this context?

Reply:
Also, `challenger_color: "black"|"white"|"automatic"|"random";`

Reply:
If one was going to do that, one would define a type like ColourSelectionOptions eh....

Reply:
Then one would realise one has to call it ColorSelectionOptions I guess :cry: